tftrt/examples/image-classification/image_classification.py

```diff
@@ -52,29 +52,34 @@ def after_run(self, run_context, run_values):
                 current_step, self.num_steps, duration * 1000,
                 self.batch_size / self.iter_times[-1]))
 
-class DurationHook(tf.train.SessionRunHook):
+class BenchmarkHook(tf.train.SessionRunHook):
     """Limits run duration"""
-    def __init__(self, target_duration):
+    def __init__(self, target_duration=None, iteration_limit=None):
         self.target_duration = target_duration
         self.start_time = None
+        self.iteration_number = 0
+        self.iteration_limit = iteration_limit
 
-    def after_run(self, run_context, run_values):
-        if not self.target_duration:
-            return
-
+    def before_run(self, run_context):
         if not self.start_time:
             self.start_time = time.time()
-            print("    running for target duration from %d" % self.start_time)
-            return
+            print("    running for target duration from %d", self.start_time)
 
-        current_time = time.time()
-        if (current_time - self.start_time) > self.target_duration:
-            print("    target duration %d reached at %d, requesting stop" % (self.target_duration, current_time))
-            run_context.request_stop()
+    def after_run(self, run_context, run_values):
+        if self.target_duration:
+            current_time = time.time()
+            if (current_time - self.start_time) > self.target_duration:
+                print("    target duration %d reached at %d, requesting stop" % (self.target_duration, current_time))
+                run_context.request_stop()
+
+        if self.iteration_limit:
+            self.iteration_number += 1
+            if self.iteration_number >= self.iteration_limit:
+                run_context.request_stop()
 
 def run(frozen_graph, model, data_files, batch_size,
-    num_iterations, num_warmup_iterations, use_synthetic, display_every=100, run_calibration=False,
-    target_duration=None):
+    num_iterations, num_warmup_iterations, use_synthetic, display_every=100, run_calibration=False, 
```

Comment:
Remove trailing whitespace.

Reply:
But this is the whitespace after comma.

Reply:
Right, but it's at the end of the line. pylint should fix this.

Reply:
This is not fixed yet.